### ISSUE TITLE
[V6] Submitting with no registered fields

### ISSUE THREAD

**Issue by mike1808** [erikras/redux-form#1231]
I have the multistep wizard form which last step has no field, just a plain submit button. 

Let's consider the follow example:

```js
function WizardFirstStep({ handleSubmit }) {
  return (
    <form onSubmit={handleSubmit}>
      <Field name="username" component="input" type="text" />
      <Field name="password" component="input" type="password" />
    </form>
  );
}

export default reduxForm({
  form: 'wizard',
  destroyOnUnmount: false,
})(WizardFirstStep);
```

```js
function WizardSecondStep({ handleSubmit }) {
  return (
    <form onSubmit={handleSubmit}>
      <button type="submit">Submit</button>
    </form>
  );
}

export default reduxForm({
  form: 'wizard',
  destroyOnUnmount: false,
})(WizardSecondStep);
```

And also the main component which changes these steps (like in the examples).

The issues is the following: in the reducer of `UNREGISTER_FIELD` action if the there is no fields `registeredFields` property is [removed](https://github.com/erikras/redux-form/blob/v6/src/reducer.js#L239) and when you try to submit with the removed property [`ReduxForm:getFieldList()`](https://github.com/erikras/redux-form/blob/v6/src/reducer.js#L239)  method will throw an error. 

I think that you shouldn't remove the `registeredFields` property from the state and let it to be an empty array.

One workaround for this is to create a hidden field `<Field name="hidden" component="input" type="hidden" />` just to make that property not to be empty.

**Comment by ooflorent** [comment]
@clayne11 could you take a look at this issue please?

**Comment by clayne11** [comment]
I'll look at it. Should be a simple fix.

**Comment by erikras** [comment]
Published fix in [`v6.0.0-rc.1`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-rc.1).
